### ISSUE TITLE
Blank or duplicate PDFs (0.8.3 or 0.8.4)

### ISSUE THREAD

**Issue by jaythrive** [dompdf/dompdf#1926]
Hello,

I'm encountering an issue where if I generate more than one PDF the second PDF is either blank or a copy of the first PDF. If a new Dompdf instance is created in the loop, it'll result in all but the first PDF being blank. If the Dompdf instance is created before the loop, it'll result in all the PDFs being duplicates of the first.

This occurs using PHP 5.6, 7.1 and 7.2.

This occurs using Dompdf 0.8.2 and 0.8.3. 

This occurs using https://github.com/mpdf/mpdf as well.

I also tried using the PDFLib https://www.pdflib.com/ library for PHP 7 instead of the PDF libraries packaged with Dompdf. If I recall, we either have to use PDFlib or it's recommended to use it for PHP 7. However, that doesn't solve the issue.

To replicate please use the following code: https://gist.github.com/jaythrive/5889a82dee5e2f23ec2fa869a145411f

The file references a "test-pdf.php" file. The code for that can be found here: https://gist.github.com/jaythrive/303b36001b3d3bdc4b42e2fb861153b3

Basically this script loops through a list of names, using the name as the PDF content and filename. This is just to demonstrate the issue.

**Comment by bsweeney** [comment]
First, you're using `require_once` in your loop, which will throw an exception on any iteration after the first. Use `require` instead.

Second, Dompdf isn't compatible with re-use within a loop right now since there's not really a "reset" option to clear out past state (see #1056). As such you'll need to re-instantiate it for each loop. Something like:

```
foreach ( $names as $name ) {
	$dompdf = new \Dompdf\Dompdf( $options );
	// populate your $html and $filename variables
	$dompdf->loadHtml( $html );
	$dompdf->render();
	$o = $dompdf->output();
	file_put_contents( $filename, $o );
	unset($dompdf);
}
```

But even this is imperfect since there are some static variables that can bleed over between runs. Better would be to do the PDF rendering in an external file and call it using `exec`.
